packages/sharing-smart-contract/contracts/modules/Subscription.sol

```diff
@@ -41,6 +47,31 @@ contract Subscription is Collection {
     /***************************************************************************
      *                        Functions                                        *
      ***************************************************************************/
+    // can subscribe for an illimited duration ?
+    function subscribeTo(uint256 _collectionId) public payable returns (uint256) {
+        require(subscriptionParams[_collectionId].duration > 0, "Subscription parameters not set");
+        require(msg.value >= subscriptionParams[_collectionId].price, "Fund sent insufficient");
+
+        uint256 extraFunds = msg.value - subscriptionParams[_collectionId].price;
+        if (extraFunds > 0) {
+            (bool success, ) = msg.sender.call{value: extraFunds}("");
+            require(success, "Failed to send back extra funds");
+        }
```

Comment:
this will trigger `msg.sender` fallback leaving some room for tricky behaviour
why not require the exact amount?

Reply:
Commit: 8c511d799ee3758caa81b46c5a47f472fa5c4126